util/ranger/detacher.go

```diff
@@ -207,7 +267,7 @@ func ExtractAccessConditions(conds []expression.Expression, rangeType RangeType,
 	case IntRangeType, ColumnRangeType:
 		return extractColumnConditions(conds, cols[0].ColName)
 	case IndexRangeType:
-		accessConds, _ := DetachIndexConditions(conds, cols, lengths)
+		accessConds, _ := DetachCNFIndexConditions(conds, cols, lengths, true)
```

Comment:
Why not `DetachIndexConditions`?

Reply:
This is used for selectivity. Selectivity cannot extract the DNF part. This will cause multiple calculation.